### PR TITLE
Manual version: Allow numeric fields to be LIKE searched

### PR DESCRIPTION
**Pull request by dereuromark**
Alternative to https://github.com/FriendsOfCake/search/pull/150 
I dont like it that this has to be done manually, but so far the autodetecting is not bullet proof and therefore this is probably the safer solution for now.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -264,11 +264,14 @@ The following options are supported by all filters except `Callback` and `Finder
...
 
-- `field` (`string|array`, defaults to the name passed to the first argument of the
+- `field` (`string|array`), defaults to the name passed to the first argument of the
   add filter method) The name of the field to use for searching. Works like the base
   `field` option but also accepts multiple field names as an array. When defining
   multiple fields, the search term is going to be looked up in all the given fields,
   using the conditional operator defined by the `fieldMode` option.
+  
+- `colType` (`array`), Use to set a custom type for any column that needs to be treated as
+  string column despite its actual type.
```
Comment (ADmad) [inline README.md]: The description could be expanded to mention that you should use this feature if you want to do "Like" searches on integer columns.

File: src/Model/Filter/Like.php [hunk 2]
```diff
@@ -56,6 +57,7 @@ public function process()
         $isMultiValue = is_array($value);
 
         $conditions = [];
+        $colTypes = $this->config('colType') ?: [];
```
Comment (ADmad) [inline src/Model/Filter/Like.php]: The `?: []` is not necessary as the default value for `colType` is already set to `[]`. Also the value can be fetched inside `if (!empty($conditions)) {` itself.

File: src/Model/Filter/Like.php [hunk 3]
```diff
@@ -78,11 +80,34 @@ public function process()
...
+            if ($colTypes) {
+                $colTypes = $this->_aliasColTypes($colTypes);
+            }
+
+            $this->query()->andWhere([$this->config('fieldMode') => $conditions], $colTypes);
         }
     }
 
     /**
+     * @param array $colTypes Column types
```
Comment (ADmad) [inline src/Model/Filter/Like.php]: Description please.

**Comment by ADmad** [comment]
Apart from the few minor things I pointed out this looks good :+1:

**Comment by dereuromark** [comment]
All done.

**Pull request merged**